### PR TITLE
Covid data hub

### PR DESCRIPTION
**Pull request by ppmarq1**
Hi! I kindly ask you for a review of this project. I have completed this milestone where we were required to:

- Install API.
- Design: Follow these [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), including:
       -  Colors, Typography: font face, size, and weight.
       -  Layout: composition and space between elements. 
       -  Original design idea by [Nelson Sakwa on Behance](https://www.behance.net/sakwadesignstudio).

- The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) was done in the README.

- Interactions

     -  Home page

    When the page loads, the web app shows the list of categories that could be filtered out by some parameter, for 
    example by the category name.  
    Along with the category name, you will display some numeric values per category that come from the API.
    When the user clicks (or taps) on a category item, the application navigates to the details page.

     -  Details page

     In the details page, the webapp retrieves data from the API to show detailed data in the category.
    When the user clicks on the "Back" button (<), the user navigates to the home page.

- Testing requirements

    - Create unit tests for pure functions in Redux code.
    - Create integration tests for application using the React Testing Library.
    - Use mock access to the API, so that your tests don't send actual requests.
    - Mock the connection to the Redux Store.

- Technical requirements

    -The project is a single-page application (SPA) built with React and Redux.
   - The data retrieved from the API should be stored in the Redux store.
   - It was filter the data that retrieve from the API using a Filter stateless component.
   - Every page (the main page and the pages for each item)  have a unique route within the SPA.
   - The project was deployed and accessible online.

- Project documentation
   - Record a 5 minutes video on Loom explaining the description of the project.
  
Kind Regards,

Pedro Marques

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,114 @@
...
+- Redux
+- API
+
+## Live Demo
+
+[Live Demo Link](https://cool-bavarois-7aa72a.netlify.app/)
+
+## Project Description
+
+Click here to watch a [video](https://www.loom.com/share/5b85672c9c3e4965b969d3bae797fdb1) that gives a brief description of the project
```
Comment (gemmen29) [inline README.md]: - [x] Please include in your video a part of your code you are proud of as per this [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md#:~:text=you%20should%20also%20highlight%20some%20interesting%20piece%20of%20code%20or%20something%20you%20built%20that%20you%20are%20very%20proud%20of.)

File: src/components/NavBar.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import { useNavigate, useLocation } from 'react-router-dom';
+import { BsMic } from 'react-icons/bs';
+import { AiOutlineSetting } from 'react-icons/ai';
+
+const Nav = () => {
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  return (
+    <nav className="navContainer">
```
Comment (gemmen29) [inline src/components/NavBar.js:11]: - [x] Kindly make your nav take the same width as the rest of the content to match the design guidelines.

| Your design | Behance design |
| --- | --- |
| image | image |

File: src/components/Home.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
+import { useState, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import CountUp from 'react-countup';
+import { FaSearch } from 'react-icons/fa';
+import { IoGlobeSharp } from 'react-icons/io5';
+import ContinentList from './ContinentList';
+
+const Home = () => {
```
Comment (gemmen29) [inline src/components/Home.js:8]: - [x] Please change the alignment of the text to the right of each element on the home page to match the design guidelines.

| Your design | Behance design |
| --- | --- |
| image | image |

File: src/components/CountryList.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import PropTypes from 'prop-types';
+import styles from './CountryList.module.css';
+
+import Country from './Country';
+
+const CountryList = ({ countries }) => (
+  <ul className="countryContainer">
```
Comment (gemmen29) [inline src/components/CountryList.js:7]: - [x] To match the design guidelines on the details page, please make only one country in the line, the country name on the left, and the count on the right.

| Your design | Behance design |
| --- | --- |
| image | image |

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @ppmarq1, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/NavBar.js (hunk 2 above)
Comment (ppmarq1) [reply src/components/NavBar.js:11]: Hi @gemmen29 , thank you for your review. In order to proceed with these changes, could you please answers my doubts below:

- May I keep the Globe symbol or should I insert a Map of the World, the same as the maps of the Continents?
- Should I  remove the search bar? This was one additional feature that I would like to include as a "code proud". But if I need to remove it, there is no problem.
- Can I keep these colors' references in my code? Behance does not offer a color guide.

Thanks for your attention!

Pedro

File: src/components/CountryList.js (hunk 4 above)
Comment (ppmarq1) [reply src/components/CountryList.js:7]: Hi @gemmen29 

   I will remove the flag by country, ok?

Thanks, 

Pedro

File: src/components/NavBar.js (hunk 2 above)
Comment (gemmen29) [reply src/components/NavBar.js:11]: Hi @ppmarq1
- Yes, you can keep the globe symbol.
- No, the search is in the requirement don't delete it and you can use it as a code you are most proud of.
- Yes, you are free to choose your main color

Comment (ppmarq1) [reply src/components/NavBar.js:11]: Thank you, I will proceed in this way.

Pedro

File: src/components/CountryList.js (hunk 4 above)
Comment (gemmen29) [reply src/components/CountryList.js:7]: Hi @ppmarq1

What I meant was to put only one country per row, not two countries, you can keep the flag but stick to the design guidelines.

Comment (ppmarq1) [reply src/components/CountryList.js:7]: Hi @gemmen29 

    Understood! My doubt was only about the flag (if remove it or not). I will put one country per row as per requirement.

Thanks,

Pedro

File: README.md [hunk 5]
```diff
@@ -1,2 +1,114 @@
...
+- Redux
+- API
+
+## Live Demo
+
+[Live Demo Link](https://cool-bavarois-7aa72a.netlify.app/)
+
+## Project Description
+
+Click here to watch a [video](https://www.loom.com/share/dd48119d67a44859aa058d493145c9e1) that gives a brief description of the project
```
Comment (medaminedev66) [inline README.md:37]: [OPTIONAL] As requested by the previous reviewer, your video should contain a highlight about a piece of the code you find interesting or are proud of making. Try to open Visual Studio Code (or whatever IDE you use) while you are recording a video and try to present a piece of code that interests you.

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @ppmarq1 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 💯 
Congratulations! 🎉 

### Highlights
- Your design looks great
- Your web app works as expected and runs without errors.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**